kernel/register.cc

```diff
@@ -649,16 +649,36 @@ void Backend::execute(std::vector<std::string> args, RTLIL::Design *design)
 {
 	std::ostream *f = NULL;
 	auto state = pre_execute();
-	execute(f, std::string(), args, design);
-	post_execute(state);
-	if (f != &std::cout)
-		delete f;
+	bool restore_log_cmd_error_throw = log_cmd_error_throw;
+	log_cmd_error_throw = true;
+	try {
+	  execute(f, std::string(), args, design);
+	  post_execute(state);
+	} catch(log_cmd_error_exception& _) {
+	  if (f && (f != &std::cout)) {
+	    delete f;
+	    f = NULL;
+
+	    for(auto& itr : filenames) {
+	      if ( itr == "<stdout>" ) continue;
+	      int code = std::remove(itr.c_str());
+	      if ( code )
+		log("\t Failed to deleted file %s - with exitcode %d\n",itr.c_str(),code);
+	    }
+	  }
+	  log_cmd_error_throw = restore_log_cmd_error_throw;
+	  if ( log_cmd_error_throw )
+	    throw log_cmd_error_exception();
+	}
+
+	log_cmd_error_throw = restore_log_cmd_error_throw;
+	if (f && (f != &std::cout)) {
+	  delete f;
+	}
```

Comment:
Please keep indentation consistent and void mixing spaces and tabs.

Reply:
ok let me redo this; thanks